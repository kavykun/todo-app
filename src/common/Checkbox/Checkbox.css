.checkbox {
  line-height: 24px;
}

input[type='checkbox'] {
  cursor: pointer;
  appearance: none;
  outline: 0;
  background: transparent;
  height: 12px;
  width: 12px;
  border: 1px solid #000;
}

input[type='checkbox']:checked {
  background: black;
}

input[type='checkbox']:after {
  content: '';
  position: relative;
  left: 40%;
  top: 20%;
  width: 15%;
  height: 40%;
  border: solid #ffff;
  border-width: 0 2px 2px 0;
  transform: rotate(45deg);
  display: none;
}

input[type='checkbox']:checked:after {
  display: block;
}

input[type='checkbox']:disabled:after {
  border-color: #ffff;
}
