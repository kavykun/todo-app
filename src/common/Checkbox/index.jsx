import React from 'react';
import PropTypes from 'prop-types';

import './Checkbox.css';

const Checkbox = (props) => {
  const { name, checked, handleInputChange } = props;

  return (
    <input
      name={name}
      type="checkbox"
      checked={checked}
      onChange={handleInputChange}
      className="checkbox"
      id="check"
    />
  );
};

Checkbox.defaultProps = {
  name: '',
  checked: false,
  handleInputChange: () => {},
};

Checkbox.propTypes = {
  name: PropTypes.string,
  checked: PropTypes.bool,
  handleInputChange: PropTypes.func,
};

export default Checkbox;
