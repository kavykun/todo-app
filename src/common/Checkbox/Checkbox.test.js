import { render, fireEvent, screen } from '@testing-library/react';

import Checkbox from './index';

test('Should call handleChange() onClick', async () => {
  const testName = 'test-checkbox';
  const handleClick = jest.fn();
  render(<Checkbox name={testName} handleInputChange={handleClick} />);
  fireEvent.click(screen.getByRole('checkbox'));
  expect(handleClick).toHaveBeenCalledTimes(1);
});

test('Should match snapshot', async () => {
  const testName = 'test-checkbox';
  const handleClick = jest.fn();
  const { asFragment } = render(
    <Checkbox name={testName} handleInputChange={handleClick} />,
  );
  expect(asFragment()).toMatchSnapshot();
});
