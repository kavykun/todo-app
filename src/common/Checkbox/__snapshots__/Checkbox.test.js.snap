// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<DocumentFragment>
  <input
    class="checkbox"
    id="check"
    name="test-checkbox"
    type="checkbox"
  />
</DocumentFragment>
`;
